utils/src/main/scala/it/pagopa/interop/commons/utils/package.scala

```diff
@@ -30,4 +31,16 @@ package object utils {
     } yield (bearerToken, correlationId, ip)
   }
 
+  def withHeaders[T](
+    f: (String, String, Option[String]) => Future[T]
+  )(implicit contexts: Seq[(String, String)]): Future[T] = extractHeaders(contexts) match {
+    case Left(ex) => Future.failed(ex)
+    case Right(x) => f.tupled(x)
+  }
+
+  def withUid[T](f: String => Future[T])(implicit contexts: Seq[(String, String)]): Future[T] =
+    AkkaUtils.getUid(contexts) match {
+      case Left(ex) => Future.failed(ex)
+      case Right(x) => f(x)
+    }
```

Comment:
The original purpose of the PR was to add just these 2 functions :trollface:

Reply:
🤣